python/pyspark/sql/dataframe.py

```diff
@@ -2045,7 +2045,46 @@ def toDF(self, *cols):
         """
         jdf = self._jdf.toDF(self._jseq(cols))
         return DataFrame(jdf, self.sql_ctx)
-
+    
+    @since(3.0)
+    def transform(self, func, *args, **kwargs):
+        """Returns a new class:`DataFrame` according to a user-defined custom transform method.
+        This allows chaining transformations rather than using nested or temporary variables.
+        
+        :param func: a user-defined custom transform function
+        :param *args: optional positional arguments to pass to `func`
+        :param **kwargs: optional keywarded arguments to pass to `func`
+        
+        A more concrete example::
```

Comment:
Let's make it to doctests

Reply:
tried to add something please review. I was not sure if I need to do imports and generate a spark context.